src/pass.h

```diff
@@ -426,8 +429,8 @@ class Pass {
 
   std::string name;
 
-  PassRunner* getPassRunner() { return runner; }
-  void setPassRunner(PassRunner* runner_) {
+  virtual PassRunner* getPassRunner() { return runner; }
+  virtual void setPassRunner(PassRunner* runner_) {
```

Comment:
Why do these two methods need to be virtual? I think we can use a fixed pass runner, even if we virtualize the passes and the addition of passes.

(If this is necessary, we'd need to benchmark it to see if this is not a general regression - it's possible that `getPassRunner` is called quite frequently during optimizations.)

Reply:
Without overriding set/get I have following assertion errors:
```
Assertion failed: (getPassRunnerAssertion failed: (getPassRunner()), function runOnFunction, fil()), function runOnFunction, file pass.h, line 481.
e pass.h, line 481.
Assertion failed: (getPassRunner()), function runOnFunction, file pass.h, line 481.
Assertion failed: (getPassRunner()), function runOnFunction, file pass.h, line 481.
```

Reply:
However I think I can workround this, by calling setPassRunner just before invoking proxied runner in case if it's pass runner is null.

Reply:
That sounds good, yes, the proxied runner can call `setPassRunner` before each proxied operation, if necessary.